Modal cohesion

kcirerick opened the pull request:
<ul>
<li>Fixed old modal bugs and layout.</li>
<li>Enable user to select a search result.</li>
<li>Query out-dated reviews servlet for all exist reviews <em>Fix in later PR!</em></li>
<li>Enable back arrow to redisplay search results.</li>
<li>Enable user to search by text, exit, search by map, select result, go back, select new result, exit, search again, and generally just navigate back and forth all they want without messing up the layout.</li>
</ul>

kcirerick commented:
Dev got a massive  update since I submitted this PR so I have a few merge conflicts to address before this PR is ready.

kcirerick commented:
Closes #69

ccjeane commented on src/main/webapp/index-script.js:
```diff
@@ -11,31 +11,56 @@ function loadPage() {
+
   const modalClosers = document.querySelectorAll('[data-modal-close-button]');
+  // Button to close the modal and deactiviate overlay
+  modalClosers.forEach(button => {
```
It seems like every PR you have been deleting/replacing these same lines of code.

ccjeane commented on src/main/webapp/index-script.js:
```diff
@@ -44,51 +69,62 @@ function loadMainButtons() {
-  
+
+  // Prevent page from refreshing when you submit the form
+  searchForm.addEventListener('submit', function(event) {
```
I'm confused why you keep moving this code.

kcirerick replied:
The first move was to decompose the loadMainButtons() function since there was a ton of code inside of that one function. Any other movements must have been me trying to fix resulting merge conflicts.

ccjeane commented on src/main/webapp/index-script.js:
```diff
@@ -117,12 +152,6 @@ function createMap() {
     mapTypeControlOptions: {mapTypeIds: ['roadmap']}});
 
-  // Create the initial InfoWindow.
-  var infoWindow = new google.maps.InfoWindow(
```
Glad you deleted this!

ccjeane commented on src/main/webapp/modal-overlay.css:
```diff
@@ -139,4 +139,36 @@ li:last-child{
+
+a:hover {
+  color: rgba(230, 230, 230); /* Grey. */ 
+<<<<<<< HEAD
```
Delete these head statements

kcirerick replied:
oops, missed a merge conflict

kcirerick commented:
> I noticed some leftover '<<< HEAD' statements and commit lines that need to be removed!
> 
> I'm also a bit confused because it seems like every other PR you have been moving around the same few lines of code. It makes reviewing these PR's really repetitive since you moved a lot of these lines already in PR #66. Perhaps try to create smaller PR's / using more branches would help eliminate some of this weirdness, and help prevent accidental code loss.

In my styleChanges PR, I moved this code out to decompose it. When I pushed modalCohesion, styleChanges had not yet been approved, so I still had the old placement of the code you are referring to. After I made the modalCohesion PR, styleChanges was approved and merged to master, which caused merge conflicts, so I moved it back, this should be the final state of that code.

dterr commented on src/main/java/com/google/sps/servlets/VotingServlet.java:
```diff
@@ -35,18 +36,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
-        review.setProperty("positive", positive + 1);
-        positive++;
+        int positive = (int) review.getProperty("positive");
+        review.setProperty("positive", ++positive);
```
Clever use of pre-increment
